py3status/modules/xscreensaver.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+"""
+Indicator for Xscreensaver, can be toggled by left clicking.
+
+This is a partial rewrite of dpms.py. Most credit
+goes to the original author Andre Doser. This module shows the status
+of Xscreensaver and activates or deactivates it upon left click.
+
+This script is useful for people who let Xscreensaver manage dpms
+settings. Xscreensaver has its own dpms variables separate from xset.
+Dpms can be savely turned of in xset as long as xscreensaver is running.
+Settings can be managed using "xscreensaver-demo".
+
+Configuration parameters:
+    - format_off: string to display when Xscreensaver is disabled
+    - format_on : string to display when Xscreensaver is enabled
+
+Example configuration in py3status.conf:
+
+```
+order += "xscreensaver"
+xscreensaver {
+    format_off = "xscreensaver off"
+    format_on = "xscreensaver on"
+}
+```
+
+@author neutronst4r <c7420{at}gmx{dot}net>
+"""
+from os import setpgrp
+from subprocess import call, Popen, DEVNULL
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    format_off = "xscreensaver off"
+    format_on = "xscreensaver on"
+    color_on = None
+    color_off = None
+
+    def xscreensaver(self, i3s_output_list, i3s_config):
+        """
+        Display a colorful state of xscreensaver.
+        """
+        self.run = call(["pidof", "xscreensaver"],
+                        stdout=DEVNULL,
+                        stderr=DEVNULL,
+                        preexec_fn=setpgrp) == 0
+        return {
+            'full_text': self.format_on if self.run else self.format_off,
+            'color': self.color_on or i3s_config['color_good']
+            if self.run else self.color_off or i3s_config['color_bad']
```

Comment:
I might move the color calculation out of the return dict to make things more readable, maybe combine with the text selection.

Also adding a `cache_until` is nice to control how often we update so not kill the cpu.

`'cached_until': time() + self.cache_timeout` is common practice with `cache_timeout` as a config option. you'd need a `from time import time` too.